File: app/src/main/java/com/feliperamoscarvalho/appconsultagithub/data/model/Pull.java
package com.feliperamoscarvalho.appconsultagithub.data.model;

import com.google.gson.annotations.SerializedName;

public class Pull {

    @SerializedName("id")
    private int id;

    @SerializedName("html_url")
    private String htmlUrl;

    @SerializedName("user")
    private User user;

    @SerializedName("title")
    private String title;

    @SerializedName("body")
    private String body;

    @SerializedName("created_at")
    private String createdAt;

    public Pull(int id, String htmlUrl, User user, String title, String body, String createdAt) {
        this.id = id;
        this.htmlUrl = htmlUrl;
        this.user = user;
        this.title = title;
        this.body = body;
        this.createdAt = createdAt;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getHtmlUrl() {
        return htmlUrl;
    }

    public void setHtmlUrl(String htmlUrl) {
        this.htmlUrl = htmlUrl;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getBody() {
        return body;
    }

    public void setBody(String body) {
        this.body = body;
    }

    public String getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(String createdAt) {
        this.createdAt = createdAt;
    }
}

File: app/src/main/java/com/joaoibarra/gitapp/api/model/Pull.kt
package com.joaoibarra.gitapp.api.model

import com.google.gson.annotations.SerializedName
import java.util.*

class Pull (
        @SerializedName("id") val id: Int,
        @SerializedName("url") val url: String,
        @SerializedName("node_id") val nodeId: String,
        @SerializedName("html_url") val htmlUrl: String,
        @SerializedName("diff_url") val diffUrl: String,
        @SerializedName("patch_url") val patchUrl: String,
        @SerializedName("issue_url") val issueUrl: String,
        @SerializedName("number") val numberId: Int,
        @SerializedName("state") val state: String,
        @SerializedName("locked") val locked: Boolean,
        @SerializedName("title") val title: String,
        @SerializedName("user") val user: User,
        @SerializedName("body") val body: String,
        @SerializedName("created_at") val createdAt: Date,
        @SerializedName("updated_at") val updatedAt: String,
        @SerializedName("closed_at") val closedAt: String,
        @SerializedName("merged_at") val mergedAt: String,
        @SerializedName("merge_commit_sha") val mergeCommitSha: String,
        @SerializedName("assignee") val assignee: User,
        @SerializedName("assignees") val assignees: List<User>,
        @SerializedName("requested_reviewers") val requestedReviewers: List<User>,
        @SerializedName("requested_teams") val requestedTeams: List<Team>,
        @SerializedName("labels") val labels: List<Label>,
        @SerializedName("milestone") val milestone: String,
        @SerializedName("commits_url") val commitsUrl: String,
        @SerializedName("review_comments_url") val reviewCommentsUrl: String,
        @SerializedName("review_comment_url") val reviewCommentUrl: String,
        @SerializedName("comments_url") val commentsUrl: String,
        @SerializedName("statuses_url") val statusesUrl: String,
        @SerializedName("head") val head: Head,
        @SerializedName("base") val base: Base,
        @SerializedName("_links") val links: Links,
        @SerializedName("author_association") val authorAssociation: String
)